raiden/blockchain/state.py

```diff
@@ -21,9 +20,11 @@ def get_channel_state(
         reveal_timeout: BlockTimeout,
         payment_channel_proxy,
         opened_block_number: BlockNumber,
-        opened_block_hash: BlockHash,
 ):
-    channel_details = payment_channel_proxy.detail(opened_block_hash)
+    # Here we have to query the latest state as during restart a lot of things
+    # can have happened, such as the channel having been closed. To get this
+    # information we should use 'latest'.
+    channel_details = payment_channel_proxy.detail('latest')
```

Comment:
Please, do not use `latest`, that is the point of having a consistent view.

`detail` does quite a few RPC calls:

- `detail_channel` get the current status of the channel 
- `detail_participant` to get the partner state
- `detail_participant` to get the current node state
- plus a few duplicated `inspect_channel`

A new block may be mined at any moment while the above calls are being done, change the status of the channel on-chain on the latest block, and the data will be corrupt. For things to be consistent we **must operate on a blockhash**

Reply:
Sure